src/events.rs

```diff
@@ -0,0 +1,53 @@
+// This file is Copyright its original authors, visible in version control
+// history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE
+// or http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
+// You may not use this file except in accordance with one or both of these
+// licenses.
+
+//! Events are surfaced by the library to indicate some action must be taken
+//! by the end-user.
+//!
+//! Because we don't have a built-in runtime, it's up to the end-user to poll
+//! [`crate::LiquidityManager::get_and_clear_pending_events()`] to receive events.
+
+use std::collections::VecDeque;
+use std::sync::{Condvar, Mutex};
+
+#[derive(Default)]
+pub(crate) struct EventQueue {
+	queue: Mutex<VecDeque<Event>>,
+	condvar: Condvar,
+}
+
+impl EventQueue {
+	pub fn push(&self, event: Event) {
+		let mut queue = self.queue.lock().unwrap();
+		queue.push_back(event);
+		self.condvar.notify_one();
+	}
+
+	pub fn pop(&self) -> Event {
+		let mut queue = self.queue.lock().unwrap();
+
+		let event = loop {
+			if let Some(event) = queue.pop_front() {
+				break event;
+			} else {
+				queue = self.condvar.wait(queue).unwrap();
+			}
+		};
+		drop(queue);
+		event
+	}
+
+	pub fn get_and_clear_pending_events(&self) -> Vec<Event> {
+		self.queue.lock().unwrap().drain(..).collect()
+	}
+}
+
+/// An Event which you should probably take some action in response to.
```

Comment:
nit:
```suggestion
/// An event which you should probably take some action in response to.
```

(alternatively `Event`)